improver/standardise.py

```diff
@@ -105,9 +105,126 @@ def grid_contains_cutout(grid, cutout):
     return True
 
 
-class StandardiseGridAndMetadata(BasePlugin):
+class StandardiseMetadata(BasePlugin):
+    """Plugin to standardise cube metadata"""
 
-    """Plugin to regrid cube data and standardise metadata"""
+    @staticmethod
+    def _collapse_scalar_dimensions(cube):
+        """
+        Demote any scalar dimensions (excluding "realization") on the input
+        cube to auxiliary coordinates.
+
+        Returns:
+            iris.cube.Cube
+        """
+        coords_to_collapse = []
+        for coord in cube.coords(dim_coords=True):
+            if len(coord.points) == 1 and "realization" not in coord.name():
+                coords_to_collapse.append(coord)
+        for coord in coords_to_collapse:
+            cube = next(cube.slices_over(coord))
+        return cube
+
+    @staticmethod
+    def _remove_scalar_coords(cube, coords_to_remove):
+        """Removes named coordinates from the input cube."""
+        for coord in coords_to_remove:
+            try:
+                cube.remove_coord(coord)
+            except CoordinateNotFoundError:
+                continue
+
+    @staticmethod
+    def _standardise_dtypes_and_units(cube):
+        """
+        Modify input cube in place to conform to mandatory dtype and unit
+        standards.
+
+        Args:
+            cube (iris.cube.Cube:
+                Cube to be updated in place
+
+        """
+
+        def as_correct_dtype(obj, required_dtype):
+            """
+            Returns an object updated if necessary to the required dtype
+
+            Args:
+                obj (np.ndarray):
+                    The object to be updated
+                required_dtype (np.dtype):
+                    The dtype required
+
+            Returns:
+                np.ndarray
+            """
+            if obj.dtype != required_dtype:
+                return obj.astype(required_dtype)
+            return obj
+
+        cube.data = as_correct_dtype(cube.data, get_required_dtype(cube))
+        for coord in cube.coords():
+            if coord.name() in TIME_COORDS and not check_units(coord):
+                coord.convert_units(get_required_units(coord))
+            req_dtype = get_required_dtype(coord)
+            # ensure points and bounds have the same dtype
+            if np.issubdtype(req_dtype, np.integer):
+                coord.points = np.around(coord.points)
+            coord.points = as_correct_dtype(coord.points, req_dtype)
+            if coord.has_bounds():
+                if np.issubdtype(req_dtype, np.integer):
+                    coord.bounds = np.around(coord.bounds)
+                coord.bounds = as_correct_dtype(coord.bounds, req_dtype)
+
+    def process(
+        self,
+        cube,
+        new_name=None,
+        new_units=None,
+        coords_to_remove=None,
+        attributes_dict=None,
+    ):
+        """
+        Perform compulsory and user-configurable metadata adjustments
```

Comment:
I think it would be useful to say briefly what the compulsory metadata adjustments are.

Reply:
OK.  For the record I moved this unchanged!  But it's still worth adding if useful.

Reply:
Thank you